torch/fx/experimental/proxy_tensor.py

```diff
@@ -914,6 +947,14 @@ def dispatch_trace(
         concrete_args: Optional[Tuple[Any, ...]] = None,
 ) -> GraphModule:
     graph = tracer.trace(root, concrete_args)
+
+    # TODO: I guess hypothetically you could have an impure node that returns
+    # a SymInt only, but between this and the catch all impure test this
+    # should be pretty obscure
+    def impure_pred(n: fx.Node) -> bool:
+        return 'val' not in n.meta or not isinstance(n.meta['val'], py_sym_types) or n.is_impure()
+
+    graph.eliminate_dead_code(impure_pred)
```

Comment:
I added this pretty late. The problem is that we actually do generate a lot of useless SymInt compute nodes in practice today, and it was a pain getting downstream to not choke on it. So kill as many of them after the fact as we can.

Reply:
Is it guaranteed that everything in the graph is functional at this point?

Reply:
No, definitely not. So we only have nodes eligible for DCE if they look like symint compute